routing/pathfind.go

```diff
@@ -617,6 +646,86 @@ func findPath(tx *bolt.Tx, graph *channeldb.ChannelGraph,
 	return pathEdges, nil
 }
 
+// findPathWithHints is a wrapper around findPath that attempts to find a path
+// to the target with the help of routing hints.
+//
+// NOTE: The hints must be chained together and sorted in forward order in order
+// to reach the target successfully.
+func findPathWithHints(tx *bolt.Tx, graph *channeldb.ChannelGraph,
```

Comment:
A set of unit test exercising this new path finding method should be added.

Reply:
Fixed.